contracts/MechMarketplace.sol

```diff
@@ -489,7 +506,154 @@ contract MechMarketplace is IErrorsMarketplace {
         IBalanceTracker(balanceTracker).finalizeDeliveryRate(msg.sender, mechDelivery.requester, requestId,
             mechDelivery.deliveryRate);
 
-        emit MarketplaceDeliver(priorityMech, msg.sender, mechDelivery.requester, requestId, requestData);
+        emit MarketplaceDeliver(priorityMech, msg.sender, mechDelivery.requester, requestId, deliveryData);
+
+        _locked = 1;
+    }
+
+    /// @dev Verifies provided request hash against its signature.
+    /// @param requester Requester address.
+    /// @param requestHash Request hash.
+    /// @param signature Signature bytes associated with the signed request hash.
+    function _verifySignedHash(address requester, bytes32 requestHash, bytes memory signature) internal view {
```

Comment:
Mostly taken from our autonolas-registries, but modified the order, such that the majority of contracts follow the standard of calling `isValidSignature()` function.